### ISSUE TITLE
Add 'hide when idle' mouse mode

### ISSUE THREAD

**Issue by dalexeev** [godotengine/godot-proposals#1099]
**Describe the project you are working on:**
A game in which there is no mouse control

**Describe the problem or limitation you are having in your project:**
I want the cursor to disappear if it remains motionless for some time, and appears when the mouse moves. `Input.MOUSE_MODE_HIDDEN` is not suitable because it hides the cursor at all.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
This mode can be selected in the Project Settings or activated in a single line of code.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
Need to add new mouse mode (`Input.MOUSE_MODE_HIDE` or `Input.MOUSE_MODE_AUTOHIDE`), delay time before cursor disappears and default mouse mode to the Project Settings.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
```
var tmr := Timer.new()

func _ready() -> void:
    tmr.one_shot = true
    tmr.wait_time = 1.2
    tmr.connect("timeout", self, "hide_cursor")
    add_child(tmr)

func _input(e: InputEvent) -> void:
    if e is InputEventMouseMotion:
        Input.set_mouse_mode(Input.MOUSE_MODE_VISIBLE)
        tmr.start()

func hide_cursor() -> void:
    Input.set_mouse_mode(Input.MOUSE_MODE_HIDDEN)
```

**Is there a reason why this should be core and not an add-on in the asset library?:**
I think this is often required in games without mouse control. Almost as often as switching to fullscreen (https://github.com/godotengine/godot/pull/39708). It makes no sense to reinvent the wheel every time. Besides, the mode that I propose is often found in other applications (for example, video players).

**Comment by Calinou** [comment]
> `Input.MOUSE_MODE_HIDDEN` is not suitable because it hides the cursor at all.

Can't you do that with a Timer node?

**Comment by dalexeev** [comment]
See description.

> Can't you do that with a Timer node?

```
extends Timer

func _ready() -> void:
    one_shot = true
    wait_time = 1.2
    connect("timeout", self, "hide_cursor")

func _input(e: InputEvent) -> void:
    if e is InputEventMouseMotion:
        Input.set_mouse_mode(Input.MOUSE_MODE_VISIBLE)
        start()

func hide_cursor() -> void:
    Input.set_mouse_mode(Input.MOUSE_MODE_HIDDEN)
```

**Comment by Calinou** [comment]
I don't think this is complex enough to warrant adding a core feature. You can do it with even less lines of code if you create the Timer node in the editor and connect it using the Node dock.

**Comment by dalexeev** [comment]
> I don't think this is complex enough

```
func _unhandled_input(_e: InputEvent) -> void:
    if Input.is_action_just_pressed('toogle_fullscreen'):
        OS.window_fullscreen = !OS.window_fullscreen
```

It's not about complexity, it's about how much it will be in demand.

**Comment by KoBeWi** [comment]
Why not just hide the cursor permanently? If the game doesn't need it, it doesn't need to be visible even for a second.

**Comment by dalexeev** [comment]
> Why not just hide the cursor permanently? If the game doesn't need it, it doesn't need to be visible even for a second.

This is the easiest option, but people don't like being deprived of control. :smile: Maybe someone wants to show some element in the game to another person.
